gbpservice/nfp/configurator/drivers/firewall/asav/asav_fw_driver.py

```diff
@@ -0,0 +1,960 @@
+        commands.append("no shutdown")
+
+        allow_inter_interface_traffic = ("same-security-traffic permit "
+                                         "inter-interface")
+        commands.append(allow_inter_interface_traffic)
+        return commands
+
+    def get_interface_position(self, mgmt_ip, mac):
+        """ Calculates the interface position.
+
+        :param mgmt_ip: Management IP address of the Service VM
+        :param mac: MAC address of the VM for which the
+        interface position has to be found.
+
+        Returns: interface index.
+
+        """
+
+        commands = list()
+
+        commands.append("sh inte")
+        result = self.configure_bulk_cli(mgmt_ip, commands,
+                                         response_data_expected=True)
+
+        if result.get('GET_RESPONSE'):
+            data = ''.join(result['GET_RESPONSE']['response']).split(
+                                                        'GigabitEthernet0/')
+            for item in data:
+                if mac in item:
+                    return item[0]
+
+    def _get_device_interface_name(self, cidr):
+        """ Prepares the interface name.
+
+        :param cidr: CIDR of the interface
+
+        Returns: interface name.
+
+        """
+
+        return 'interface-' + cidr.replace('/', '_')
+
+    def configure_interfaces(self, context, resource_data):
+        """ Configures interfaces for the service VM.
+
+        :param context: neutron context
+        :param resource_data: a dictionary of firewall rules and objects
+        send by neutron plugin
+
+        Returns: SUCCESS/Failure message with reason.
+
+        """
+
+        try:
+            mgmt_ip = resource_data['mgmt_ip']
+            provider_ip = resource_data.get('provider_ip')
+            provider_cidr = resource_data.get('provider_cidr')
+            provider_mac = resource_data.get('provider_mac')
+            stitching_ip = resource_data.get('stitching_ip')
+            stitching_cidr = resource_data.get('stitching_cidr')
+            stitching_mac = resource_data.get('stitching_mac')
+            provider_interface_position = resource_data.get(
+                                    'provider_interface_index')
+            stitching_interface_position = resource_data.get(
+                                    'stitching_interface_index')
+
+            (provider_mac, stitching_mac) = self.get_asav_macs(
+                [provider_mac, stitching_mac])
+
+            provider_interface_position = self.get_interface_position(
+                                                        mgmt_ip, provider_mac)
+            stitching_interface_position = str(int(
+                                            provider_interface_position) + 1)
+
+            provider_macs = [provider_mac]
+            stitching_macs = [stitching_mac]
+        except Exception as err:
+            msg = ("Failed before issuing a configure interfaces"
+                   " call. Error: %r." % err)
```

Comment:
Correct the log message. Mention why it failed rather than which position it failed. Say "Failed to calculate interface position ...."